### PR TITLE
ACS-306: ACS Repository: V1 Rest API for content direct access urls

### PR DESCRIPTION
**Pull request by cristianturlica**
- added request content url (direct access) to nodes, nodes versions, nodes renditions and nodes versions renditions
   - added tests

### CODE REVIEW

File: src/main/java/org/alfresco/rest/api/impl/NodesImpl.java [hunk 1]
```diff
@@ -245,6 +247,8 @@ public void setNonAttachContentTypes(Set<String> nonAttachWhiteList)
         this.nonAttachContentTypes = nonAttachWhiteList;
     }
 
+    private int directAccessUrlExpiryTime = -1;
```
Comment (gavincornwell) [inline src/main/java/org/alfresco/rest/api/impl/NodesImpl.java:250]: Personally I think we should have a default value.

File: src/main/java/org/alfresco/rest/api/impl/NodesImpl.java [hunk 2]
```diff
@@ -3360,6 +3369,33 @@ public Node unlock(String nodeId, Parameters parameters)
...
+            throw new InvalidArgumentException("Missing nodeRef");
+        }
+
+        // get configured expiry time and convert to milliseconds
+        int expiryTime = directAccessUrlExpiryTime > 0 ? (directAccessUrlExpiryTime * 1000) : -1;
+        String directAccessUrl = contentService.getDirectAccessUrl(nodeRef, expiryTime);
+
+        if (StringUtils.isEmpty(directAccessUrl))
+        {
+            throw new DisabledServiceException("Direct access url isn' available.");
```
Comment (gavincornwell) [inline src/main/java/org/alfresco/rest/api/impl/NodesImpl.java:3393]: Typo, "isn'" -> "isn't"

File: src/main/java/org/alfresco/rest/api/nodes/NodesEntityResource.java [hunk 3]
```diff
@@ -1,193 +1,203 @@
...
-
-    @Operation("lock")
-    @WebApiDescription(title = "Lock Node",
-            description="Places a lock on a node.",
-            successStatus = HttpServletResponse.SC_OK)
-    public Node lock(String nodeId, LockInfo lockInfo, Parameters parameters, WithResponse withResponse)
-    {
-        return nodes.lock(nodeId, lockInfo, parameters);
-    }
+/*
```
Comment (gavincornwell) [inline src/main/java/org/alfresco/rest/api/nodes/NodesEntityResource.java:1]: Most of the file is showing as changed when only a few lines have been added!?

File: src/main/java/org/alfresco/rest/api/impl/NodesImpl.java (hunk 2 above)
Comment (cristianturlica) [reply src/main/java/org/alfresco/rest/api/impl/NodesImpl.java]: Done

File: src/main/java/org/alfresco/rest/api/impl/NodesImpl.java (hunk 1 above)
Comment (cristianturlica) [reply src/main/java/org/alfresco/rest/api/impl/NodesImpl.java]: I added the requested 24 hours as the default

File: src/main/java/org/alfresco/rest/api/nodes/NodesEntityResource.java (hunk 3 above)
Comment (cristianturlica) [reply src/main/java/org/alfresco/rest/api/nodes/NodesEntityResource.java:1]: Some old files have a different line ending and is automatically changed to the default of the repository

File: src/test/java/org/alfresco/rest/api/tests/client/data/NodeDirectAccess.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    public void expected(Object o)
+    {
+        assertTrue("o is an instance of " + o.getClass(), o instanceof NodeDirectAccess);
+
+        NodeDirectAccess other = (NodeDirectAccess) o;
+
+        AssertUtil.assertEquals("contentUrl", getContentUrl(), other.getContentUrl());
+    }
+
+    public JSONObject toJSON()
```
Comment (ancutam) [inline src/test/java/org/alfresco/rest/api/tests/client/data/NodeDirectAccess.java:51]: `toJSON()` and `parseEntity()` don't seem to be used anywhere.

Comment (cristianturlica) [reply src/test/java/org/alfresco/rest/api/tests/client/data/NodeDirectAccess.java]: Good spot, I removed them.

File: src/main/java/org/alfresco/rest/api/model/DirectAccessUrlRequest.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+ * #L%
+ */
+package org.alfresco.rest.api.model;
+
+import java.util.Date;
+
+public class DirectAccessUrlRequest
+{
+    private Date expiresAt;
+    private Long validFor;
```
Comment (DenisGabriela) [inline src/main/java/org/alfresco/rest/api/model/DirectAccessUrlRequest.java:33]: We could add a javadoc to this class to specify what `validFor` is representing, e.g. `seconds` or something else.

**Pull request merged**